kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -293,6 +290,12 @@ maybeTermAnd notSimplifier childTransformers first second = do
             lift $ Builtin.KEqual.unifyKequalsEq childTransformers notSimplifier unifyData
         | Just unifyData <- Builtin.KEqual.matchUnifyKequalsEq second first =
             lift $ Builtin.KEqual.unifyKequalsEq childTransformers notSimplifier unifyData
+        | Just unifyData <- Builtin.Int.matchUnifyIntEq first second =
+            lift $ Builtin.Int.unifyIntEq childTransformers notSimplifier unifyData
+        | Just unifyData <- Builtin.Int.matchUnifyIntEq second first =
+            lift $ Builtin.Int.unifyIntEq childTransformers notSimplifier unifyData
+        | Just unifyData <- Builtin.String.matchUnifyStringEq first second = lift $ Builtin.String.unifyStringEq childTransformers notSimplifier unifyData
+        | Just unifyData <- Builtin.String.matchUnifyStringEq second first = lift $ Builtin.String.unifyStringEq childTransformers notSimplifier unifyData
```

Comment:
`matchUnifyStringEq` and `matchUnifyIntEq` should handle matching their arguments both ways. That is, this should just be:

```suggestion
        | Just unifyData <- Builtin.Int.matchUnifyIntEq first second =
            lift $ Builtin.Int.unifyIntEq childTransformers notSimplifier unifyData
        | Just unifyData <- Builtin.String.matchUnifyStringEq first second = lift $ Builtin.String.unifyStringEq childTransformers notSimplifier unifyData
```

`matchUnifyKequalsEq` should do the same, actually. These functions are the canonical way to match these unification problems and they should be symmetric with respect to their arguments.